compiler/src/dmd/expressionsem.d

```diff
@@ -15218,7 +15218,9 @@ extern(C++) Expression toLvalue(Expression _this, Scope* sc)
 }
 
 // e = original un-lowered expression for error messages, in case of recursive calls
-private Expression toLvalueImpl(Expression _this, Scope* sc, Expression e) {
+private Expression toLvalueImpl(Expression _this, Scope* sc, Expression e)
+{
+    assert(e);
```

Comment:
```suggestion
{
    assert(e);
    Expression makeANameUp = e ? e : _this.e1;
    // or 
    e = e ? e : _this.e1
```
Reads a lot better than repetition down below.

Reply:
Actually I'm a dummy

Reply:
I don't want this function to accept 'nullable' classes, the caller should pass proper instances. Later I'd like to refactor `modifiableLvalue` to so it doesn't accept a `null` parameter either.